src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,231 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car fro DB by car id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+            return Optional.of(car);
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT c.id, c.model, c.manufacturer_id, m.name, m.country "
+                + "FROM cars c "
+                + "JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllCarsStatement = connection.createStatement()) {
+            ResultSet resultSet = getAllCarsStatement
+                    .executeQuery(query);
+            while (resultSet.next()) {
+                Car car = getCarFromResultSet(resultSet);
+                cars.add(car);
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars from DB", e);
+        }
+        if (cars.size() > 0) {
+            for (Car car: cars) {
+                car.setDrivers(getDriversForCar(car.getId()));
+            }
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(3, car.getId());
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update car in DB " + car, e);
+        }
+        deleteDriversByCar(car);
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT c.id, c.model, c.manufacturer_id, cd.driver_id, "
+                + "d.name AS driver_name, d.license_number, m.name, m.country "
```

Comment:
```suggestion
        String query = "SELECT c.id, c.model, c.manufacturer_id, cd.driver_id, "
                + "d.name AS driver_name, d.license_number, m.name, m.country "
```

why do we need `driver_name` and `license_number` here?